### PR TITLE
Add QuantizedDot to ngraph core

### PR DESCRIPTION
**Pull request by nbpatel**

### CODE REVIEW

File: src/ngraph/builder/quantized_dot_builder.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
+//*****************************************************************************
+// Copyright 2018-2019 Intel Corporation
```
Comment (silee2) [inline src/ngraph/builder/quantized_dot_builder.cpp]: Use `2017-2019`

Comment (silee2) [inline src/ngraph/builder/quantized_dot_builder.cpp]: Use `2017-2019`

File: src/ngraph/builder/quantized_dot_builder.hpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
+//*****************************************************************************
+// Copyright 2018-2019 Intel Corporation
```
Comment (silee2) [inline src/ngraph/builder/quantized_dot_builder.hpp]: Use `2017-2019`

File: src/ngraph/op/quantized_dot.cpp [hunk 3]
```diff
@@ -0,0 +1,219 @@
+//*****************************************************************************
+// Copyright 2018-2019 Intel Corporation
```
Comment (silee2) [inline src/ngraph/op/quantized_dot.cpp]: Use `2017-2019`

File: src/ngraph/op/quantized_dot.hpp [hunk 4]
```diff
@@ -0,0 +1,76 @@
+//*****************************************************************************
+// Copyright 2018-2019 Intel Corporation
```
Comment (silee2) [inline src/ngraph/op/quantized_dot.hpp]: Use `2017-2019`

File: src/ngraph/op/quantized_dot.cpp [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+    // AxisSet should be empty till we support channel wise quantization
+    NODE_VALIDATION_CHECK(this,
+                          m_input0_axes == AxisSet{} && m_input1_axes == AxisSet{} &&
+                              m_output_axes == AxisSet{},
+                          "Input0, input1 and output AxisSet should be empty");
+
+    const PartialShape& arg0_shape = get_input_partial_shape(0);
+    const PartialShape& arg1_shape = get_input_partial_shape(1);
+    PartialShape result_shape;
+    size_t m_reduction_axes_count = 1;
```
Comment (aprocter) [inline src/ngraph/op/quantized_dot.cpp]: Why is this hard-wired to 1? This is inconsistent with the non-quantized `Dot`.

File: src/ngraph/runtime/cpu/mkldnn_emitter.hpp [hunk 6]
```diff
@@ -277,6 +277,7 @@ namespace ngraph
 
                     mkldnn::post_ops ops;
 
+#if 0
```
Comment (sayantan-nervana) [inline src/ngraph/runtime/cpu/mkldnn_emitter.hpp]: an `#if 0` here. maybe we can get rid of the code there, unless it serves some purpose?

File: src/ngraph/runtime/cpu/pass/cpu_mkldnn_primitive_build.cpp [hunk 7]
```diff
@@ -2400,7 +2400,7 @@ namespace ngraph
                     writer << "*cg_ctx->mkldnn_descriptors[" << desc_index + 2 << "]);\n";
 
                     writer << "\nmkldnn::post_ops ops;\n";
-                    if (has_relu<OP>(node))
+                    if (false)
```
Comment (sayantan-nervana) [inline src/ngraph/runtime/cpu/pass/cpu_mkldnn_primitive_build.cpp]: similar to the`#if 0`, here is some relu related code, which is dead-code right now

File: src/ngraph/frontend/onnx_import/utils/matmul_factory.cpp [hunk 8]
```diff
@@ -174,34 +174,77 @@ std::shared_ptr<ngraph::Node>
     QLinearMatmulFactory::make_dot(const std::shared_ptr<ngraph::Node>& left,
                                    const std::shared_ptr<ngraph::Node>& right)
 {
```
Comment (diyessi) [inline src/ngraph/frontend/onnx_import/utils/matmul_factory.cpp]: Change these to Output<Node>

File: src/ngraph/runtime/cpu/cpu_emitter.cpp [hunk 9]
```diff
@@ -2594,21 +2593,21 @@ namespace ngraph
...
+                       << " , " << out[0].get_type() << ", int32_t>(" << args[0].get_name()
+                       << ",\n";
                 writer << "            " << args[1].get_name() << ",\n";
                 writer << "            " << out[0].get_name() << ",\n";
                 writer << "            {" << join(args[0].get_shape()) << "},\n";
                 writer << "            {" << join(args[1].get_shape()) << "},\n";
                 writer << "            {" << join(out[0].get_shape()) << "},\n";
                 writer << "            1,\n";
-                writer << "            dyn_scale);\n";
+                writer << "            " << args[2].get_name() << ",\n";
```
Comment (diyessi) [inline src/ngraph/runtime/cpu/cpu_emitter.cpp]: You might want `*args[2]` here and below; it gives more information.

File: src/ngraph/frontend/onnx_import/utils/matmul_factory.cpp (hunk 8 above)
Comment (nbpatel) [reply src/ngraph/frontend/onnx_import/utils/matmul_factory.cpp]: Thats onnx bridge code which hasn't been touched in this PR. The change you are mentioning needs to happen at a lot of other places as well in onnx bridge.

**Comment by tsocha** [comment]
This PR will be affected by #3330 because `matmul_factory` will be moved to builders.

**Pull request merged**